electrum/address_synchronizer.py

```diff
@@ -970,6 +976,10 @@ def get_utxos(
         if excluded_addresses:
             domain = set(domain) - set(excluded_addresses)
         mempool_height = block_height + 1  # height of next block
+        cache_key = ''.join(domain) + f"{mature_only}{confirmed_funding_only}{confirmed_spending_only}{nonlocal_only}{block_height}"
```

Comment:
I would prefer clear separators for clarity.
While it cannot happen in today's Bitcoin, imagine if "aaa" and "bbb" and "aaabbb" were all valid addresses. You should really specify if you are caching "aaa,bbb" or "aaabbb".
This is why the cache_key in get_balance was designed as such.
```suggestion
        cache_key = sha256(
            ','.join(sorted(domain))
            + f";{mature_only};{confirmed_funding_only};{confirmed_spending_only};{nonlocal_only};{block_height}"
        )
```

Reply:
Yeah, fair cop, the full cache key is apparently just under 0.5M here so the SHA256 (~0.1ms) is worth it; applied